add generic alm-example to olm-catalog 0.1.3 placeholders

komish opened the pull request:
Nearly there, @raffaelespazzoli - thanks so much for your help resolving [this](https://bugzilla.redhat.com/show_bug.cgi?id=1794821)!

The gitdiff reflects that I created new files, but understand that I simply took the directory structure for the `olm-catalog/resource-locker-operator/0.1.2` and copied it over to `0.1.3`, and then made the following changes:

1. Bumped all version references to `0.1.3`
2. In the **ClusterServiceVersion**, I added the **Secret** object to the permission list. That will translate to a the **Role** which is given to the operator's **ServiceAccount** being updated when installed via OLM in OpenShift. [Link](https://github.com/redhat-cop/resource-locker-operator/pull/19/files#diff-3f8b290ae0a146a1e01b7192f1759214R202)
3. In the **ClusterServiceVersion**, I updated all version references from previous `0.1.2` to `0.1.3`.
4. In the **ClusterServiceVersion**, I removed the original `test-simple-resource` CR and replaced it with one that can properly be created out-of-the-box with no changes required by the end user. [Link](https://github.com/redhat-cop/resource-locker-operator/pull/19/files#diff-3f8b290ae0a146a1e01b7192f1759214R8)

The **CustomResourceDefinition** has not been changed.

### What needs to be done
This merely stages the bundle data that's needed to update the operator in https://github.com/operator-framework/community-operators but the images still need to be built, published, and everything needs to be merged over there as well whenever you have time!

### Goal of this PR
Final piece necessary to fix this [bug](https://bugzilla.redhat.com/show_bug.cgi?id=1794821). 🎉 

Allows an end-user to create an instance of a **ResourceLocker** successfully. It uses the operator's default **ServiceAccount** (`resource-locker-operator`) which is not a best practice, but should be ok for demonstration purposes. That **ServiceAccount** has the ability to create secrets with this PR, as well as list them at the cluster scope which is a requirement (this piece was already in place, no change happens here re: the deployed **ClusterRole** or the `clusterPermissions` CSV spec item).

raffaelespazzoli commented:
we have to do this differently. this has to do with the process needed to release to OLM.
Please add the examples you'd like to see in the `alm-examples` field as files with extension .cr and format yaml in the `./deploy/crds` directory. The release process will take care of putting them in the csv.

komish commented:
@raffaelespazzoli Sure thing, though I see the CR that's already included is a placeholder. I can translate all of the `alm-examples` to CR files. Let me know if you want this done differently.

raffaelespazzoli commented:
feel free to delete the placeholder CR, I should have done it a long ago and add your more meaningful examples.

komish commented:
@raffaelespazzoli Done! I've split the alm-examples into CRs that get properly parsed when you `generate` a csv using operator-sdk.

I've also updated the roles yaml file to include the added **Secrets** permission.

Finally, since the project has bumped to operator-sdk 0.16.0, I've updated the documentation on generating catalog components to deprecate use of `olm-catalog gen-csv` in favor of `generate csv`

 Let me know if any further changes are required.

raffaelespazzoli commented on deploy/role.yaml:
```diff
@@ -31,6 +31,7 @@ rules:
   resources:
   - configmaps
   - pods
+  - secrets
```
why do we need this?

komish replied:
@raffaelespazzoli Currently, the included example CRs do not install due to permissions issues. From what I can see, in order to create a locked resource the service account that's used needs to be able to list/get/watch at a cluster scope as well as create it at the namespace scope (if it's a namespaced resource. 

I changed the example CR to create a **ResourceLocker** that locks a **Secret** because we already have Get/List/Watch for **Secrets** at the cluster scope. The only thing missing was the ability to create secrets at the namespace scope. 

I thought about instead adding Get/List/Watch for **ConfigMaps** at the cluster scope instead (since we already have the ability to create them at the namespace) and then having the demo CR be a **ConfigMap**, but adding cluster scoped permissions seemed like too much. 

Ultimately, the goal is as stated in the original post ("**Goal of this PR**") - allow for an end user to install the included example successfully.

raffaelespazzoli replied:
I don't like the idea of having to add permissions just to run the examples.
Any other example that would work without adding permissions? Or in alternative would it be ok for you to add a few instructions on prerequisites needed to run the examples?

komish replied:
@raffaelespazzoli 

> I don't like the idea of having to add permissions just to run the examples.

Yep, that's totally fair! The only justification I found acceptable was that the service account already had what is effectively "reader" privileges for secrets at the cluster scope so adding "writer" privileges at the namespace_seemed_ less impactful. But you make a good point, it's purely in order to run the example. 

> Any other example that would work without adding permissions?

I tested a few, but couldn't find any that logically functioned out of the box.

> Or in alternative would it be ok for you to add a few instructions on prerequisites needed to run the examples?

We could do this if it's not already there (I do think there's something there already but I would need to double check). Either way, pre-requisite instruction means that the example CR will never build successfully without executing those (which could be missed by users or automation). I'm OK with this trade-off, but it will mark the bugzilla bug I've referenced as WONTFIX. 

Totally your call here! If you want to remove the permission, I'll go ahead and submit that change here, and explain in bugzilla the circumstance and the implications of the design (where the operator has less privilege, relying on the referenced service account to have the privilege needed) to close the bug.

Let me know!

komish commented:
@raffaelespazzoli friendly bump!
